File: 412-fizz-buzz/412-fizz-buzz.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        fizzbuz = []
        for i in range(1, n+1):
            if i%3==0 and i%5==0:
                fizzbuz.append('FizzBuzz')
            elif i%3==0 and i%5!=0:
                fizzbuz.append('Fizz')
            elif i%3!=0 and i%5==0:
                fizzbuz.append('Buzz')
            else:
                fizzbuz.append(str(i))
        return fizzbuz

File: 412-fizz-buzz/412-fizz-buzz.scala
import scala.collection.mutable.ListBuffer

object Solution {
    def fizzBuzz(n: Int): List[String] = {
        var list = new ListBuffer[String]()
        def go(start: Int, end: Int): List[String] = {
            if(start % 3 == 0 && start % 5 == 0) list += "FizzBuzz"
            else if(start % 3 == 0) list += "Fizz"
            else if(start % 5 == 0) list += "Buzz"
            else list += start.toString
            if(start == end) list.toList
            else go(start+1, end)
        }
        go(1, n)
    }
}